tool/build.dart

```diff
@@ -200,19 +231,26 @@ _main(List<String> argv) async {
       }
     }
 
-    var cr = await exec("git", args: [
+    var gitArgs = <String>[
       "clone",
-      "--depth=1",
-      repo,
-      "."
-    ], writeToBuffer: true);
+      "--depth=1"
+    ];
+
+    if (branch != "HEAD") {
+      gitArgs.add("--branch=${branch}");
```

Comment:
Here, if the branch requested is not `HEAD`, we will add `--branch` to the clone command to build from the given branch.